### PR TITLE
Consolidate and update KKP installation instructions

### PR DESCRIPTION
**Pull request by embik**
Our installation instructions for both master and seeds needed some love, so I worked on consolidating existing information and updating instructions to match KKP behaviour changes that were recently introduced.

I realize this became quite a big PR, but here's the key changes:

1. Remove the duplicate "Install on {AWS,Azure,GCE,vSphere}" pages. While I understand what they were going for, the content changes on each page were minimal but created an absolutely unreadable navigation under the "Installation" menu item.
2. All installation pages have gained provider-specific tabs in places where we can provide provider-specific information, mirroring the previously removed pages' content into a single guide. We also do this for KubeOne.
3. The instructions for using start.kubermatic have been removed; the service is barely maintained and a bad choice for getting into KKP at this point.
4. A couple of minor typos and outdated information pieces have been updated across several pages.
5. Seed setup instructions have been updated extensively. Parts of it is restructuring existing content, but some information was outdated and/or not accurate anymore, so I've spent some time on:
    * Removing outdated instructions (e.g. CRDs no longer need to be applied manually), improving the instructions around cluster backup configuration (marking the MinIO setup as optional, explaining how to set up cluster backup locations, etc).
    * Adding bits and pieces that were missing (e.g. passing the `--storageclass` flag to the installer, or creating a job that creates a bucket in MinIO for backups).
    * We also give some light examples for datacenter configurations now, which was adapted from the provider-specific seed installation instructions.
6. Remove the duplicate seed setup instructions for EE and instead mirror what we do for the master setup instructions (link to the CE and mention the changes from that).

### CODE REVIEW

**Comment by embik** [comment]
/hold

Putting a hold on this to get a couple of comments before merging.

**Comment by embik** [comment]
You can find the preview for the installation section [available here](https://deploy-preview-1288--cranky-newton-4e6ed2.netlify.app/kubermatic/main/installation/).

File: content/kubermatic/main/installation/install-kkp-CE/_index.en.md
Comment (xmudrii) [inline content/kubermatic/main/installation/install-kkp-CE/_index.en.md:393]: Since this URL is not actually working, I'd put it in a code block.

```suggestion
With all this in place, you should be able to access `https://kubermatic.example.com/` (i.e. the URL to your KKP setup that you
```

File: content/kubermatic/main/installation/install-kkp-CE/add-seed-cluster-CE/_index.en.md
Comment (xmudrii) [inline content/kubermatic/main/installation/install-kkp-CE/add-seed-cluster-CE/_index.en.md:10]: I'd add a note that this expects KKP to be already installed and link to the original installation document.

File: content/kubermatic/main/installation/install-kkp-CE/add-seed-cluster-CE/_index.en.md
Comment (xmudrii) [inline content/kubermatic/main/installation/install-kkp-CE/add-seed-cluster-CE/_index.en.md:184]: I think we should highlight that you should follow option 3 in this case.

File: content/kubermatic/main/installation/install-kkp-CE/add-seed-cluster-CE/_index.en.md
Comment (xmudrii) [inline content/kubermatic/main/installation/install-kkp-CE/add-seed-cluster-CE/_index.en.md:275]: We should clarify if you need to follow this if you have shared master/seed setup.

File: content/kubermatic/main/installation/install-kkp-CE/add-seed-cluster-CE/_index.en.md
Comment (xmudrii) [inline content/kubermatic/main/installation/install-kkp-CE/add-seed-cluster-CE/_index.en.md:459]: We can mention that if you used different bucket name previously, make sure to update it here.

**Review by xmudrii — APPROVED** [review]
What an amazing PR! Great job! :tada: :heart: 

/lgtm
/approve

**Pull request merged**
